doc/how_to/streamlit_migration/chat.md

```diff
@@ -0,0 +1,341 @@
+
+Below we will show you how to build a custom `ChatInput` widget.
+
+```python
+import param
+
+import panel as pn
+
+pn.extension(design="material")
+
+
+class ChatInput(pn.viewable.Viewer):
+    value = param.String()
+
+    disabled = param.Boolean()
+    max_length = param.Integer(default=5000)
+    placeholder = param.String("Send a message")
+
+    def __init__(self, **params):
+        layout_params = {
+            key: value
+            for key, value in params.items()
+            if not key in ["value", "placeholder", "disabled", "max_length"]
+        }
+        params = {
+            key: value for key, value in params.items() if key not in layout_params
+        }
+
+        super().__init__(**params)
+
+        self._text_input = pn.widgets.TextInput(
+            align="center",
+            disabled=self.param.disabled,
+            max_length=self.param.max_length,
+            name="Message",
+            placeholder=self.param.placeholder,
+            sizing_mode="stretch_width",
+        )
+        self._submit_button = pn.widgets.Button(
+            align="center",
+            disabled=self.param.disabled,
+            icon="send",
+            margin=(18, 5, 10, 0),
+            name="",
+            sizing_mode="fixed",
+        )
+        pn.bind(
+            self._update_value,
+            value=self._text_input,
+            event=self._submit_button,
+            watch=True,
+        )
+
+        self._layout = pn.Row(
+            self._text_input, self._submit_button, align="center", **layout_params
+        )
+
+    def __panel__(self):
+        return self._layout
+
+    def _update_value(self, value, event):
+        self.value = value or self.value
+        self._text_input.value = ""
+
+
+chat_input = ChatInput(placeholder="Say something")
+
+
+@pn.depends(chat_input.param.value)
+def message(prompt):
+    if not prompt:
+        return ""
+    return f"User has sent the following prompt: {prompt}"
+
+
+pn.Column(message, chat_input, margin=50).servable()
+```
+
+![Panel ChatInput](../../_static/images/panel_chat_input.png)
+
+## Chat Status
+
+### Streamlit Chat Status
+
+```python
+import time
+import streamlit as st
+
+with st.status("Downloading data...", expanded=True):
+    st.write("Searching for data...")
+    time.sleep(1.5)
+    st.write("Downloading data...")
+    time.sleep(1.5)
+    st.write("Validating data...")
+    time.sleep(1.5)
+
+st.button("Run")
+```
+
+![Streamlit status](https://user-images.githubusercontent.com/42288570/275434382-992f352f-676a-4167-aad0-1fcc2745c130.gif)
+
+### Panel Chat Status
+
+Panel does not provide a dedicated *status* component because it is built into Panels high-level `ChatInterface`. Furthermore Panel provides a lot of other built in *indicators*. Check out the [Indicators Gallery](https://panel.holoviz.org/reference/index.html#indicators).
+
+Below we will show you how to build a custom `Status` indicator.
+
+```python
+import time
```

Comment:
Should we add a `Status` component like this to Panel? We should consider whether we need an extra "ready" `value` option: How do we want to show this status if we have not yet started a run. Should is be disabled, not visible or have some custom look?

Reply:
Maybe object and objects would be more in line with Panels kaminhylden convention than step and steps?